// graph-component/index.js
const cytoscape = require('cytoscape');

module.exports = function(){
  return component;
}

function component( container, componentState ){

  const root = document.createElement("div");
  root.style.height = "100%";
  root.style.width = "100%";
  root.style.display = "block";
  container.element.appendChild(root);


container.on( 'open', function(){
    var cy = cytoscape({
      container: root,

      motionBlur: true,
      motionBlurOpacity: 0.2,

      elements: [ // list of graph elements to start with
          { // node a
            data: { id: 'a' }
          },
          { // node b
            data: { id: 'b' }
          },
          { // edge ab
            data: { id: 'ab', source: 'a', target: 'b' }
          }
        ],

        style: [ // the stylesheet for the graph
          {
            selector: 'node',
            style: {
              'background-color': '#666',
              'label': 'data(id)'
            }
          },

          {
            selector: 'edge',
            style: {
              'width': 3,
              'line-color': '#ccc',
              'target-arrow-color': '#ccc',
              'target-arrow-shape': 'triangle',
              'curve-style': 'bezier'
            }
          }
        ],

        layout: {
          name: 'grid',
          rows: 1
        }









    });


  })
    // global.cy = cy;
    //
    // var collection = cy.collection();
    // cy.nodes().on('click', function(e){
    //   var clickedNode = e.target;
    //
    //   collection = collection.union(clickedNode);
    //   console.log(collection);
    // });
    //
    // var eles = cy.add([
    //   { group: 'nodes', data: { id: 'n0' }, position: { x: 100, y: 100 } },
    //   { group: 'nodes', data: { id: 'n1' }, position: { x: 200, y: 200 } },
    //   { group: 'edges', data: { id: 'e0', source: 'n0', target: 'n1' } }
    // ]);
    //
    // var b = cy.$('#b');
    // cy.remove( b );

};
